Translation of latest updates to French (#90)
QuiltMC/developer-wiki: wiki/configuration/getting-started/fr.md

# Débuter avec Quilt Config

[Quilt Config](https://github.com/QuiltMC/quilt-config) est la puissante API de configuration de Quilt. Elle permet aux developpeur·euse·s (c'est vous !) de facilement mettre en place des valeurs qui peuvent être modifiées par l'utilisateur·ice pour changer le comportement de votre mod. Elle est inclue dans le [Quilt Loader](https://github.com/QuiltMC/quilt-loader) pour permettre à n'importe quel mod Quilt de l'utiliser sans importer de nouvelles dépendences ou lancer de setup.

## Mettre en place votre config

Dans ce tutorial, nous allons constuire un mod simple qui écrit une ligne de texte dans la console quand il est initilisé. N'oubliez pas que c'est un exemple et votre mod peut utiliser les valeurs de la config n'importe où ! Vous pouvez voir le code source complété de ce tutoriel [sur GitHub](https://github.com/ix0rai/quilt-config-example-mod).

<!-- TODO: Add code project into wiki -->

> Pour débuter avec Quilt Config, vous aurez besoin d'une classe dédiée pour contenir votre config. Pour un mod extrémement configurable, vous pourriez vouloir séparer votre config dans plusieurs fichiers dans un package `config` mais pour ce tutoriel et la majorité des mods nous allons tout mettre dans un seul fichier.

Reviewer: ```suggestion
Pour débuter avec Quilt Config, vous aurez besoin d'une classe dédiée pour contenir votre config. Pour un mod extrémement configurable, vous pourriez vouloir séparer votre config dans plusieurs fichiers dans un package `config`, mais pour ce tutoriel et la majorité des mods nous allons tout mettre dans un seul fichier.
```

> Créons un nouveau fichier pour notre config, dans le même dossier que notre 'mod initializer'. Nous l'appelerons `ExampleModConfig`, mais vous pouvez l'appeler comme vous voulez. Cette nouvelle classe de configuration hérite de l'API `ReflectiveConfig` qui fournit tout ce dont nous avons besoin pour faire fonctionner la config.

Reviewer: ```suggestion
Créons un nouveau fichier pour notre config, dans le même dossier que notre `ModInitializer`. Nous l'appelerons `ExampleModConfig`, mais vous pouvez l'appeler comme vous voulez. Cette nouvelle classe de configuration hérite de l'API `ReflectiveConfig` qui fournit tout ce dont nous avons besoin pour faire fonctionner la config.
```

- Ensuite, nous appelons la méthode `QuiltConfig.create(String family, String id, Class<C> configCreatorClass)`. Comme vous pouvez le voir elle prend trois paramètres :
  - Le string `family`, qui indique le dossier dans lequel votre config sera stockée, par rapport au dossier de configuration global (en général `<your instance directory>/config`). Nous utiliserons l'ID de notre mod comme dossier et c'est une bonne pratique.
  - Un deuxième string `id`, qui sera le nom du fichier de configuration (avant que l'extension de fichier soit ajouté, ce qui change en fonction du format que vous choisissez). Ici nous utilisons à nouveau l'ID de notre mod mais si vous avez une config compliqué avec plusieurs fichiers vous voudrez utiliser différents noms.
  - Enfin, quelque-chose de bizarre : c'est quoi un `Class<C> configCreatorClass` ? On ne va pas trop s'occuper des détails : passez simplement le nom de votre classe de configuration avec `.class` ajouté à la fin, comme on l'a fait ici avec `ExampleModConfig.class`.

## Ajouter des valeurs

> Voilà ! Nous avons maintenant une config à laquelle nous pouvons accéder partout dans notre projet. Le problème étant, on n'a aucune raison d'y accéder parce qu'il n'y y a rien. Réglons ce problème en ajoutant une valeur ! Notre mod d'exemple écrit une ligne de texte dans la console au démarage : pourquoi pas permettre à l'utilisateur·ice de choisir quelle est cette ligne. Ajoutons un deuxième champ dans notre config :

Reviewer: ```suggestion
Voilà ! Nous avons maintenant une config à laquelle nous pouvons accéder partout dans notre projet. Le problème étant, on n'a aucune raison d'y accéder parce qu'il n'y a rien. Réglons ce problème en ajoutant une valeur ! Notre mod d'exemple écrit une ligne de texte dans la console au démarage : pourquoi pas permettre à l'utilisateur·ice de choisir quelle est cette ligne. Ajoutons un deuxième champ dans notre config :
```

- Cette valeur est `public final`, ce qui signifie qu'au lieu d'être accessible par n'importe quoi il faut d'abord une instance de `ExampleModConfig`. Notez qu'il _faut_ utiliser `public final` quand vous créer une valeur de config en utilisant Quilt Config. Vu qu'on a déjà défini notre champ `INSTANCE`, nous pourrons cette valeur n'importe où via `ExampleModConfig.INSTANCE.message`.
- Le type est `TrackedValue<String`. Les chevrons (`<>`) nous permettent d'utiliser des _generics_, une méthode d'adapter une classe à votre besoin spécifique en Java. Ici, la classe `TrackedValue` nous permet d'adapter le type d'objet qu'elle contient, donc nous l'utilisons pour contenir un `String`. Grâce aux generics, nous pourrions mettre un autre type dans ces chevrons plus tard pour contenir un autre type de valeur ! C'est du foreshadowing. Nous allons stocker d'autres valeurs. Soyez prêt·e·s.
- Nous appelons la méthode `value`, qui vient de la classe `ReflectiveConfig` dont on hérite. Cette méthode prend un paramètre, qui sera la valeur par défaut d'une valeur de configuration. Ici, l'autrice originale de ce tutoriel l'utilise pour pour faire sa promo.

> Ouf. On s'en est sorti ! C'était le dernier gros afflux d'informations de ce tutoriel et on peut commencer à s'amuser un peu. Maintenant qu'on a un champ dans notre config, on peut retourner à notre 'mod initializer' et commencer à l'utiliser !

Reviewer: ```suggestion
Ouf. On s'en est sorti ! C'était le dernier gros afflux d'informations de ce tutoriel et on peut commencer à s'amuser un peu. Maintenant qu'on a un champ dans notre config, on peut retourner à notre `ModInitializer` et commencer à l'utiliser !
```

    // ...
    @IntegerRange(min = 0, max = 10)
    public final TrackedValue<Integer> timesToPrint = this.value(1);
}
```

Nous n'allons pas montrer les changements dans le code pour utiliser chacune de ces valeurs mais n'oubliez pas que vous pouvez voir comment elles sont utilisées dans le [code terminé](https://github.com/ix0rai/quilt-config-example-mod). Revenons à nos listes à puces pour expliquer cet extrait !

- `IntegerRange` est une annotation : elle ajoute un nouvelle fonctionnalité à notre champ, au délà de la déclaration normale. `IntegerRange` nous permet de limiter les valeurs possibles de notre champ `Integer`, pour qu'elles soient contenues entre `min` et `max` (de manière inclusive).
- Nous avons changé le type de `String` à `Integer`. Pourquoi `Integer` au lieu de `int` comme nous l'utiliserions pour un champ numérique normal ? `int` est un _type primitif_, ce qui signifie que ce n'est pas une classe ! Comme les generics ne peuvent prendre que des classes, Java fournit des classes pour chaque primitive. `boolean` devient `Boolean`, `float` devient `Float`, `double` devient `Double`, `char` devient `Character`, etc, etc.

> Quelque chose de primordial à ne pas oublier est que vous ne pouvez pas simplement mettre n'importe quelle classe dans les generics de `TrackedValue` : Quilt Config doit savoir comment le sérialiser. Par défaut, tous les [types primitifs](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html) sont sérialisables ainsi que la classe `String`. Nous expliquerons comment dire à Quilt Config comment sérialiser une classe plus tard mais d'abord : plus de fun avec des annotations !

Reviewer: "primordial" est un grand mot pour utiliser ici :p
(pas un problème, c'est juste drôle)
